### ISSUE TITLE
Some chord names are incorrect

### ISSUE THREAD

**Issue by Monochrome37** [googlecreativelab/chrome-music-lab#1]
I'm using the Chords section of the Chrome Music Lab with Firefox 44.0.2 on Windows 10.

When I click on any of the lowest eight notes, the correct chord name appears above. However when I click on G#/Ab it also says G major chord. It says G major chord for the four adjacent notes from G# to B. Then when I click middle C it says C major chord and the rest are correct.

I've attached a screenshot of an example.
![chord lab example](https://cloud.githubusercontent.com/assets/17742294/13633527/8b979100-e5e7-11e5-8a8d-705a2ea810d2.png)

Thanks.

**Comment by cwilso** [comment]
+1.  Just noticed this.

**Comment by Cyclone975** [comment]
Yep, specifically, the G#, A, A# and B keys all show G as the root note, in both octaves, for both major and minor types. 

If this is to be educational, this really should be fixed.

Also, perhaps consider a Sharp/Flat mode so the flat names can be displayed for kids learning all the chord names. Right now it only shows the sharp names.

**Comment by mathewray-google** [comment]
Thanks for the head's-up, we've corrected the issue for English and will be correcting any other affected languages shortly.

**Comment by Tchanders** [comment]
+1 @Cyclone975

The issue surrounding sharp keys vs. their flat equivalents may seem subtle, but some of the sharp keys included in this app will look very strange to anyone who has studied any music theory.

Naming the chord below "A sharp major" means that the three notes highlighted are technically A sharp, C double sharp and E sharp. Calling it "B flat major" would mean that the notes are B flat, D and F. The key A sharp major is almost never used in notated music (see https://en.wiktionary.org/wiki/A-sharp_major and https://www.basicmusictheory.com/a-sharp-major-scale).

It may seem overly fussy, but anyone who reads music to more than a very basic level will notice it straight away and it detracts from the professionalism of this otherwise awesome app!

The major and minor keys that musicians recognise are summarised here: https://en.wikipedia.org/wiki/Key_signature#Notational_conventions

![screen shot 2016-03-11 at 13 29 15](https://cloud.githubusercontent.com/assets/10440647/13703404/d58748bc-e78d-11e5-8b9c-b6cf9045d755.png)

**Comment by alexanderchen** [comment]
Hello! Thanks for the feedback, and glad you like the experiment. We did give this some thought.

Overall our hope was to have this experiment be more of a lightweight introduction to chords, where you focus less about the spelling of the chords. The goal is to grasp the concept of a three note chord visually, and see how by switching one note, you can change its entire feeling from major to minor.

We agree that A# Major, with its double-sharp and E# make it more uncommon, and that Bb Major is much more common. We considered looking at each chord and using the most common spelling. But it introduced complexities. For example, if we switched to the most common spelling of each chord, we would use C# minor, but Dflat major. And that means that to people who aren't familiar with the concept of chords, when they flip the bottom switch from major to minor, would suddenly see it switch letters, from C# to Dflat. And it might draw more attention to that spelling change than the simple concept of going from major to minor. And so we decided to prioritize learning about the "major" / "minor" switch.

We also considered putting a global toggle button to switch all spellings from their flat to sharp counterparts, but again, it seemed to add complexity that might detract and draw too much attention to the spelling.

But I agree there's probably not a one-size-fits-all for learning about chords. And we are open to hearing how music teachers and students use this tool. We did make this one completely open source in the hopes that anyone who has new ideas for variations can help us build new tools. 

Alex

**Comment by Cyclone975** [comment]
Hi Alex, thanks for the reply. I really love the simplicity of the interface. 
As a music teacher myself, I consider the sharp/flat issue pretty important and fundamental to learning chords. Here is a scenario I can imagine a kid needing to toggle between them and in the process learning a lot about chords - I imagine a kid sitting down with a lead sheet or a chord book... let's say it's a Beatles song book... and they are trying to follow along and teach themselves how the chords fit onto piano. Say the song is in Eb. This kid is going to have a tough time getting through the 3 chords most likely to be on that lead sheet... Eb Major, Ab Major and Bb Major. They could search through all the keys and never find the right chords from the book. That's why I think a simple little (♭ / #) toggle would be really pedagogically sound. While I understand not wanting to draw too much attention to spellings, I think kids would have a chance to teach themselves intuitively about enharmonics, which could really be a big deal to kids working their way through theory.
Thanks for your consideration of this feature, and thanks for your work on this great little app.

**Comment by alexanderchen** [comment]
Hey - that's really helpful feedback, and especially valuable as you are a music teacher. A flat/sharp toggle is indeed probably the simplest solve for the kind of scenario you're describing. We're taking a step back, a bit of pause, before adding any new features, as most of our team is taking a moment to regroup after our big launch of these experiments. But we've made a note of this idea, and are keeping an eye out for feedback from folks before diving back in and making any major feature updates. Thanks for the really thoughtful feedback.

**Comment by DurableMicron** [comment]
I can also vouch for this, when you're taught music theory they teach you key signatures/scales starting with C, and then adding one sharp or flat each time. So after C you learn G,(F#), and F(Bb). The idea of calling Bb A# while it may not seem like a big thing, is actually a concept that some kids can struggle to take a while to understand.
